go/integration/awskms/aws_kms_aead.go

```diff
@@ -32,7 +33,7 @@ import (
 // AWSAEAD represents a AWS KMS service to a particular URI.
 type AWSAEAD struct {
 	keyURI string
-	kms    *kms.KMS
+	kms    kmsiface.KMSAPI
```

Comment:
Could you please explain why kmsiface.KMSAPI is better than kms.KMS?

Reply:
Hi @thaidn. I don't know if there is other way to provide the mocked kms client to awskms.AWSAEAD struct.  
 
There is an example how to use mocks with kms at the bottom page at https://docs.aws.amazon.com/sdk-for-go/api/service/kms/kmsiface/. I used it as reference. 

I believe it's possible to misconfigure somehow kms.KMS client to get instant error on `Decrypt` call but I decided to focus on mock client.